server.js

```diff
@@ -1,29 +1,91 @@
+  },
+});
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+const documentation = {
+  "Welcome": "Welcome to Claire's Happy Thoughts API 🌼",
+  "GET endpoint": {
+    "https://claireshappythoughts.herokuapp.com/thoughts": "Use this endpoint in your application to show up to the last 20 thoughts posted",    
+  },  
+  "POST endpoint 1": {
+    "https://claireshappythoughts.herokuapp.com/thoughts": "Use this endpoint in your application to allow for the user to post a thought to the database. Only allows for the message to be sent in the body of the post request.",
+  },
+  "POST endpoint 2": {
+    "https://claireshappythoughts.herokuapp.com/thoughts/:thoughtId/like": " Use this endpoint to allow for users to like thoughts by including the thought id in the url which targets the thought with that specific id and then updates the heart count of that thought in the database.",
+  },
+};
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.json(documentation);
+});
+
+// thoughts get route which will show a max of 20 thoughts, sorted by createdAt to show the most recent 20 thoughts first 
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20);
+  // returning a good status code and the json object that we saved to the database
+  res.status(200).json(thoughts);
+});
+
+// thoughts post route which will be used to post new thoughts to the database
+app.post('/thoughts', async (req, res) => {
+  // Try catch form, but you can also use promises (then with a catch). This will try to save the new thought and if it doesn't succeed, for instance if it doesn't meet the min and maxlength defined in the model, then the error will be shown.
+  try {
+    // create new thought with the data passed via the body of the post request from the client side via the API endpoint and then save that to the database. Only the message can be updated and not the whole object as message is the only property specified and saved to the database
+    const thought = await new Thought({message: req.body.message}).save();
+    // returning a good status code and the json object that we saved to the database
+    res.status(200).json(thought);
+  } catch (error) {
+    // returning a bad error code and a message with some more information to the user as to why the thought wasn't sent.
+    res.status(400).json({message: "Could not save new thought to the database. Please make sure that you've entered a valid thought that is between 5-140 letters long.", errors:error.errors})
+  }
```

Comment:
I am not understand how this is work, I am sure that is good, so if you have time, can you explain this for me?(: Great work!

Reply:
@lilihorvath Do you meant the error message of the new thought isn't successful?